### PR TITLE
fixed loading custom losses and metrics for load_model

### PR DESCRIPTION
**Pull request by PhilippeNguyen**
the `convert_custom_objects` function currently only used
to deserialize losses and metrics in `load_model` cannot handle
custom losses and metrics that are formatted in the
same conventions as other keras objects.
Changed to directly call `deserialize` from the appropriate
module and allow for the case that the losses and metrics
can be a list of deserializable keras objects.

This will allow for the writing of losses as keras objects/ classes with get_config() . Can be executed with the __call__ function, see constraints.py. By allowing this kind of object, loss functions can have extra attributes outside of the function parameters `y_true` and `y_pred`.
Also matches the deserialization of all other keras objects.

### CODE REVIEW

**Comment by fchollet** [comment]
Thanks for the PR. It isn't clear what issue this is meant to fix, so please add a unit test showing the purpose of the fix (it should be a test of a use case that would fail at `master` but would pass with the fix).

**Comment by PhilippeNguyen** [comment]
Looks like it's been fixed by #8056
